Postgres test fails - UTC timezone

### Preflight Checklist

- [x] I agree to follow the [Code of Conduct](https://github.com/dexidp/dex/blob/master/.github/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/dexidp/dex/issues) for an issue that matches the one I want to file, without success.
- [x] I am not looking for support or already pursued the available [support channels](https://github.com/dexidp/dex/issues/new/choose) without success.

### Version

2.44.0

### Storage Type

Postgres

### Installation Type

Binary

### Expected Behavior

`$ export DEX_POSTGRES_HOST="127.0.0.1"`

```
$ make testall
...
PASS
```

### Actual Behavior

`$ export DEX_POSTGRES_HOST="127.0.0.1"`

```
$ make testall
...
=== RUN   TestPostgres/DeviceRequestCRUD
    conformance.go:1015: 
        	Error Trace:	/home/user/repos/dex/storage/conformance/conformance.go:1015
        	            				/home/user/repos/dex/storage/conformance/conformance.go:31
        	Error:      	Not equal: 
        	            	expected: storage.DeviceRequest{UserCode:"WPDP-KLRP", DeviceCode:"ottfednbgoqsgxee3svxcgtrh", ClientID:"client1", ClientSecret:"secret1", Scopes:[]string{"openid", "email"}, Expiry:time.Date(2125, time.December, 12, 14, 11, 18, 0, time.UTC)}
        	            	actual  : storage.DeviceRequest{UserCode:"WPDP-KLRP", DeviceCode:"ottfednbgoqsgxee3svxcgtrh", ClientID:"client1", ClientSecret:"secret1", Scopes:[]string{"openid", "email"}, Expiry:time.Date(2125, time.December, 12, 14, 11, 18, 0, time.Location("Etc/UTC"))}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -12,3 +12,28 @@
        	            	   ext: (int64) 67056819078,
        	            	-  loc: (*time.Location)(<nil>)
        	            	+  loc: (*time.Location)({
        	            	+   name: (string) (len=7) "Etc/UTC",
        	            	+   zone: ([]time.zone) (len=1) {
        	            	+    (time.zone) {
        	            	+     name: (string) (len=3) "UTC",
        	            	+     offset: (int) 0,
        	            	+     isDST: (bool) false
        	            	+    }
        	            	+   },
        	            	+   tx: ([]time.zoneTrans) (len=1) {
        	            	+    (time.zoneTrans) {
        	            	+     when: (int64) -9223372036854775808,
        	            	+     index: (uint8) 0,
        	            	+     isstd: (bool) false,
        	            	+     isutc: (bool) false
        	            	+    }
        	            	+   },
        	            	+   extend: (string) (len=4) "UTC0",
        	            	+   cacheStart: (int64) -9223372036854775808,
        	            	+   cacheEnd: (int64) 9223372036854775807,
        	            	+   cacheZone: (*time.zone)({
        	            	+    name: (string) (len=3) "UTC",
        	            	+    offset: (int) 0,
        	            	+    isDST: (bool) false
        	            	+   })
        	            	+  })
        	            	  }
        	Test:       	TestPostgres/DeviceRequestCRUD
...
FAIL
make: *** [Makefile:131: testrace] Error 1
```

### Steps To Reproduce

_No response_

### Additional Information

_No response_

### Configuration

```yaml

```

### Logs

```shell

```

--- a/storage/conformance/conformance.go
+++ b/storage/conformance/conformance.go
@@ -1225,6 +1225,14 @@ func testDeviceRequestCRUD(t *testing.T, s storage.Storage) {
 		t.Fatalf("failed to get device request: %v", err)
 	}
 
+	// Storage backends are not expected to preserve the time.Location of the
+	// stored value, only the instant. Postgres, for example, returns the
+	// expiry in a loaded "Etc/UTC" location rather than the time.UTC
+	// singleton, which makes reflect.DeepEqual (used by require.Equal)
+	// report a difference even though the two times are the same instant.
+	// Normalize to UTC before comparing, matching the other CRUD tests.
+	got.Expiry = got.Expiry.UTC()
+
 	require.Equal(t, d1, got)
 
 	// No manual deletes for device requests, will be handled by garbage collection routines

